IQSS/dataverse-frontend
# Collection permissions check

**What this PR does / why we need it**:

- Checks user permissions to create a Collection & create a Dataset on a specific collection, if the user doesn't have permissions neither to create a collection or a dataset, the Add Data dropdown inside the Collection page will not appear; if it has permissions to create a collection or dataset it will be visible with the proper buttons enabled or disabled in it.
- Checks user permissions to create a Collection inside the Create Collection page.
- Checks user permissions to create a Dataset inside the Create Dataset page.
- Check user permissions on the root collection to enable or disabled buttons inside the Add Data drop-down located in the application header.

**Which issue(s) this PR closes**:

Closes #434 

**Special notes for your reviewer**:
I have also changed the alias 'root' used in several components for an exported constant, to avoid mistakes and improve reusability, as it is done in js-dataverse.
E2E test `navigates to the collection page after submitting a valid form` is failing, will be ok after merging [PR 169](https://github.com/IQSS/dataverse-client-javascript/pull/169) from js-dataverse and also [PR 451](https://github.com/IQSS/dataverse-frontend/pull/451) from this repo.

**Suggestions on how to test this**:

1. Create a Collection and a Dataset in the JSF version with User A, publish both Collection and Dataset.
2. With User A, Navigate to that collection and check that you are able to see the Add Data dropdown with both the New Collection and New Dataset buttons enabled in it, click on both buttons and check that you can successfully the create dataset form and the create collection form.
3. Create another user "B" in the JSF version.
4. Navigate in the SPA to the Collection created from user A, and the Add Data dropdown shouldn't be visible.
5. Log in with user "A" and navigate to the created Collection in JSF, click Edit => Permissions, go to Users Groups and assign roles to user "B" as Dataset Creator to be able to add datasets to that Collection.
6. Log in with user "B" and navigate to that collection in the SPA and check that in the Add Data dropdown you have only the New Dataset button enabled.
7. Log out from user B, Log in with user A and navigate again to that Collection and assign a different role (Curator for example), so user B will be able to Add Dataset and Add Collection.
8. Again log in with user A and navigate to that collection in the SPA and check that in the Add Data dropdown you have both the New Collecction and the New Dataset buttons enabled.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:
<img width="1004" alt="Screenshot 2024-08-16 at 15 56 54" src="https://github.com/user-attachments/assets/741818b0-36cf-4677-9656-bf998800805c">
<img width="998" alt="Screenshot 2024-08-16 at 15 56 30" src="https://github.com/user-attachments/assets/9f389148-fa36-450d-89c7-0b7dcfe67256">


**Is there a release notes update needed for this change?**:
No

**Additional documentation**:
No
